doc/nrf/migration/migration_guide_1.x_to_2.x.rst

```diff
@@ -0,0 +1,430 @@
+(usually subelement of the ``.addr`` element), and a few other attributes.
+
+The ``cbor_read_object`` takes the specification and, using the TinyCBOR decoding context, attempts to get all the described fields into the specified designated variables.
+After a successful call to ``cbor_read_object``, the variables are set and ready for further processing, and the buffer is shifted beyond the list.
+This also includes copying binary and string buffers to the specified locations.
+
+The context, accessed by ``ctxt->it`` in the above example, is provided by the MCUmgr library, so it is already initialized and has the buffer attached.
+
+The zcbor conversion of the above code looks like this:
+
+.. code-block:: c
+
+   int image;
+   struct zcbor_string zst;
+   struct zcbor_map_decode_key_val image_upload_decode[] = {
+           ZCBOR_MAP_DECODE_KEY_VAL(image, zcbor_int32_decode, &image),
+           ZCBOR_MAP_DECODE_KEY_VAL(data, zcbor_bstr_decode, &zst),
+   };
+   ok = zcbor_map_decode_bulk(zsd, image_upload_decode,
+                              ARRAY_SIZE(image_upload_decode), &decoded) == 0;
+   ...
+
+The list specification is significantly simplified as it consists of array of ``ZCBOR_MAP_DECODE_KEY_VAL`` defined mappings, each given: 
+
+ * Name of a key (without quotation marks).
+ * zcbor decoding function directly from API (or user-provided type equivalent).
+ * Pointer to a variable where the extracted value will be stored.
+
+The difference here is that the decoding of binary and string data does not involve copying of the data to a specified buffer.
+Instead, as described in :ref:`migration_tinycbor_to_zcbor_general`, the decoding function is paired with a ``struct zcbor_string`` type variable where decoding will store position and length of the data, within the provided CBOR buffer.
+It is then up to the user to copy the values to a specified location.
+This mechanism improves the RAM and CPU usage because the need for an intermediate buffer is often reduced.
+
+To process a CBOR buffer and decode it according to the defined mapping, ``zcbor_map_decode_bulk`` takes the following parameters:
+
+ * The zcbor decoding context, ``zsd`` in the above example.
+ * The mapping specification and size of the mapping.
+ * A pointer to a variable for number of successfully decoded map elements.
+
+.. note::
+   Currently, there is no method provided to find out what fields did not get found in string.
+   It is up to the user to verify whether keys that are optional have appeared in the decoded string.
+   This can be done, for example, by checking if a destination variable value has changed from initial.
+
+New Zephyr logging service - logging v2
+***************************************
+
+Zephyr v3.x.x comes with a new logging service (logging v2) that replaces and extends the older one.
```

Comment:
It is more that Zephyr v3.x.x is using new logging (logging v2) by default. Legacy version is still supported but it is deprecated and will be removed after Zephyr v3.1.0. It can be enabled by setting `CONFIG_LOG1=y`.